src/rank.py

```diff
@@ -267,6 +267,8 @@ def getScoreSVM(self, comp, entry):
             value = entry[at]
             if isinstance(value, str):
                 value = value.lower()
+            if comp.operation == 'AND':
+                return 0 # TODO - Review why did this come here. Special case after fixing Unit class usage. 
```

Comment:
Is AND a possible operation here? It has been a while since I have looked at this code, but I would think that the boolean operations would be handled somewhere else.
If AND is a possible operation here, should we also check for and return 0 on operations such as OR and NOT?

Reply:
AND is not valid but the code was entering here and failing.

Reply:
That didn't fully answer what I was asking. Why did you not introduce code to handle OR or NOT? If AND is a possible operation at that location, why was it not caught earlier?
Also, I will not accept a PR which is clearly in progress. If you did what the TODO comment you wrote says, then I would consider adding your fix. If for whatever reason you don't want to do that, then you can keep your change on your own branch, but I am not merging it to main.
Thanks!